.github/workflows/update_changelog.yml

```
@@ -0,0 +1,136 @@
+  update-changelog:
+    if: needs.changelog-entry.outputs.optIn
+    runs-on: ubuntu-latest
+    needs: changelog-entry
+    steps:
+      - name: Check if PR exists
+        id: check
+        continue-on-error: true
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          prs=$(gh pr list \
+              --repo "$GITHUB_REPOSITORY" \
+              --json title \
+              --label "changelog" \
+              --jq 'length')
+          if [[  $prs -gt 0  ]]; then
+              echo "existing=true" >> "$GITHUB_OUTPUT"
+          fi
+      - uses: actions/checkout@v4
+      - name: check for branch
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        continue-on-error: true
+        run: |
+          BRANCH=automated-changelog
+          if gh api repos/hashicorp/terraform-provider-azurerm/branches/$BRANCH > /dev/null 2>&1; then
+            echo "Branch exists on remote..."
+            git fetch origin $BRANCH
+            git checkout $BRANCH
+          else
+            echo "Branch does not exist on remote, creating locally..."
+            git checkout -b $BRANCH
+          fi
+          
+      - name: Create pull request
+      #if changelog PR isn't already open, open one
+      #create a new PR, start with appending the release number and (unreleased)
+        if: '!steps.check.outputs.existing'
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        continue-on-error: true
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+          
+          #new pull request for new release needs the headers all added to the top
+          FILE="CHANGELOG.md"
+          version=$(head -n 1 "$FILE")
+          IFS='.' read major minor patch <<< "$version"
+          ((minor++))
+          patch=$(echo $patch | sed 's/ (.*)//')
+          new_version="${major}.$minor.${patch} (Unreleased)"
+          headers="${new_version}\n\nENHANCEMENTS:\n\nFEATURES:\n\nBUG FIXES:\n"
+          temp_file=$(mktemp)
+          echo -e "$headers" > "$temp_file"
+          cat "$FILE" >> "$temp_file"
+          mv "$temp_file" "$FILE"
+          echo "File has been updated."
+
+          major=$(echo $major | sed 's/## //')
+          RELEASENUM="${major}.$minor.${patch}"
+
+          git add CHANGELOG.md
+          git commit -m "staring new changelog PR"
+          git push --set-upstream origin automated-changelog
+          echo "Creating a new pull request"
+          gh pr create \
+          --repo "$GITHUB_REPOSITORY" \
+          --base main \
+          --head automated-changelog \
+          -l "changelog" \
+          -t "CHANGELOG.md for $RELEASENUM" \
+          -b "Automated changelog for next release, $RELEASENUM"
+      
+      - name: Set up Go
+        uses: actions/setup-go@v3  # Set up go
+        with:
+          go-version: '1.20'
```

The rest of the tooling in the provider _should_ be on 1.22 currently
```suggestion
          go-version: '1.22'
```